### DISCUSSION TITLE
rasterio error on windows

### DISCUSSION THREAD

**Discussion by zacforsman** [cnr-isti-vclab/TagLab#31] [General]
Installed in windows 11: 
`PS C:\Users\zfors\TagLab-main> python.exe --version
Python 3.7.9
PS C:\Users\zfors\TagLab-main> nvcc --version
nvcc: NVIDIA (R) Cuda compiler driver
Copyright (c) 2005-2019 NVIDIA Corporation
Built on Wed_Oct_23_19:32:27_Pacific_Daylight_Time_2019
Cuda compilation tools, release 10.2, V10.2.89`

Install script completed with no errors, Taglab starts but several tools don't work (4 point segmentation, add or remove points).  I get the following error code:

`TagLab 2022.02.25
C:\Users\zfors\AppData\Roaming\Python\Python37\site-packages\rasterio\__init__.py:226: NotGeoreferencedWarning: Dataset has no geotransform set. The identity matrix may be returned.
  s = DatasetReader(path, driver=driver, sharing=sharing, **kwargs)`

**Comment by zacforsman** [comment]
OK, I found the dependancies.md file (should probably mention this in the documentation).... also I think there should be a bit more documentation on the order to install things ... doesn't gdal need to be installed before rasterio?   -struggling for 3 days so far to install properly with cuda....whew!

**Reply by zacforsman** [reply]
No luck, after carefully following the dependancies.md document, I get the same error...
## windows with dependancies.md; shell as admin
`python.exe --version`
Python 3.7.9
`nvcc --version`
nvcc: NVIDIA (R) Cuda compiler driver
Copyright (c) 2005-2019 NVIDIA Corporation
Built on Wed_Oct_23_19:32:27_Pacific_Daylight_Time_2019
Cuda compilation tools, release 10.2, V10.2.89

if previously installed, remove versions...uninstall all pip (run as admin)
`pip list`
 To copy the packages name to a file `pip freeze > requirements.txt`
 To delete all packages `pip uninstall -r requirements.txt -y`
 Check all packages are removed `pip list`

**[windows binaries](https://www.lfd.uci.edu/~gohlke/pythonlibs/)**.

download binary then:
`pip install --force-reinstall .\GDAL-3.4.1-cp37-cp37m-win_amd64.whl`
`pip install .\rasterio-1.2.10-cp37-cp37m-win_amd64.whl`

checked on torch webpage (dependancies.md)for proper version
`pip install torch==1.10.2+cu102 torchvision==0.11.3+cu102 torchaudio===0.10.2+cu102 -f https://download.pytorch.org/whl/cu102/torch_stable.html`

`python3 install.py` #slowed down considerably during pytorch but no errors

close window and restart shell, `python TagLab.py` error!
C:\Program Files\Python37\lib\site-packages\rasterio\__init__.py:226: NotGeoreferencedWarning: Dataset has no geotransform set. The identity matrix may be returned.
  s = DatasetReader(path, driver=driver, sharing=sharing, **kwargs)

**Reply by maxcorsini** [reply]
Hi Zac. This not seems an error but a Warning raised by rasterio. TagLab starts but you cannot add images ? What is the format of the orthoimages that you use ? Please, let us know.
Regarding the 4-points clicks tool and the positive-negative clicks tool taking into account that to put a point you have to press Shift together with the left mouse button. In the next days we will release the updated documentation.

**Reply by zacforsman** [reply]
AAAhhh! Thank you!  
I can add images....they are in jpg format. I didn't know you had to click shift and thought the screen was frozen (now I feel quite dumb). I don't remember clicking Shift for the cpu version so I was confused...

On the positive side, I can confirm that TagLab works in win11, and with the new sublayer for linux the GUI launches with no issues without the need for Xserv... pretty slick!

-Thanks again for the quick response.

-Zac
